Automattic/wp-calypso
# Marketplace: Add search header

#### Changes proposed in this Pull Request

* Adds a header to the marketplace.
* Moves the search component from the navigation bar to the header.
* Adds a list of recommended searches.

#### Screenshots
| Mobile | Tablet | Desktop |
|--------|-------|----------|
|<img width="322" alt="Screen Shot 2022-03-23 at 4 20 29 PM" src="https://user-images.githubusercontent.com/1035546/159797876-ab430d31-4575-44dc-964e-e7381db74ef5.png">|<img width="769" alt="Screen Shot 2022-03-23 at 4 20 09 PM" src="https://user-images.githubusercontent.com/1035546/159797921-ec758e37-87ab-4e4f-a1e1-03333e990488.png">|<img width="1512" alt="Screen Shot 2022-03-23 at 4 19 30 PM" src="https://user-images.githubusercontent.com/1035546/159797959-0045100c-b56c-4e68-a118-b52af55f093d.png">|

#### Testing instructions

* Navigate into `http://calypso.localhost:3000/plugins`.
* It should render the heading, the search bar and recommended searches.
* Write `seo` on the search bar it should show `seo` related plugins.
* Click the `portfolio` recommended search, it should show `portfolio` related plugins.
* Navigate into `http://calypso.localhost:3000/plugins/:siteSlug`.
* It should render the heading, the search bar and recommended searches.
* Write `seo` on the search bar it should show `seo` related plugins.
* Click the `portfolio` recommended search, it should show `portfolio` related plugins.




Related to #61863
